cmd/spiffe-helper/config.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/hashicorp/hcl"
+	"github.com/spiffe/spiffe-helper/pkg/helper"
+)
+
+// ParseConfig parses the given HCL file into a SidecarConfig struct
+func ParseConfig(file string) (helperConfig *spiffe.HelperConfig, err error) {
+	helperConfig = &spiffe.HelperConfig{
+		AgentAddress:       `hcl:"agentAddress"`,
+		Cmd:                `hcl:"cmd"`,
+		CmdArgs:            `hcl:"cmdArgs"`,
+		CertDir:            `hcl:"certDir"`,
+		SvidFileName:       `hcl:"svidFileName"`,
+		SvidKeyFileName:    `hcl:"svidKeyFileName"`,
+		SvidBundleFileName: `hcl:"svidBundleFileName"`,
+		RenewSignal:        `hcl:"renewSignal"`,
+		Timeout:            `hcl:"timeout"`,
+	}
+
+	// Read HCL file
+	dat, err := ioutil.ReadFile(file)
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse HCL
+	if err := hcl.Decode(helperConfig, string(dat)); err != nil {
+		return nil, fmt.Errorf("unable to decode configuration: %v", err)
+	}
+	if err != nil {
```

Comment:
This check can be removed.

Reply:
I had to change back to hcl.Parse() and hcl.DecoeObject() because hcl.Decode does not remove the `hcl:...` tags from the struct if the field is not populated in the config file.